ShoppingCart-topic1/src/Offer.java

```diff
@@ -0,0 +1,56 @@
+import java.util.ArrayList;
+
+public class Offer implements Item {

```

Comment:
I used here the Composite pattern. Because I think that an Offer is an Item (has a name, code, value) who contains other items (it can be products or even other offers).